_posts/2019-10-16-deployment-updates.md

```diff
@@ -0,0 +1,214 @@
+efforts, and provides links to ongoing work.
+
+What we mean by Deployment
+--------------------------
+
+In order to run Dask on a cluster, you need to setup a scheduler on one
+machine:
+
+```
+$ dask-scheduler
+Scheduler running at tcp://192.168.0.1
+```
+
+And start Dask workers on many other machines
+
+```
+$ dask-worker tcp://192.168.0.1
+$ dask-worker tcp://192.168.0.1
+$ dask-worker tcp://192.168.0.1
+$ dask-worker tcp://192.168.0.1
+```
+
+For informal clusters people might do this manually, by logging into each
+machine and running these commands themselves.  However it's much more common
+to use a cluster resource manager such as Kubernetes, Yarn (Hadoop/Spark),
+SGE, PBS, SLURM, LSF, some cloud service or some custom system.
+
+As Dask is used by more institutions and used more broadly within those
+institutions making deployment smooth and natural becomes increasingly
+important.  This is so important in fact, that there have been seven separate
+efforts to improve deployment in some regard or another by a few different
+groups.
+
+We'll briefly summarize and link to this work below, and then we'll finish up
+by talking about some internal design that helped to make this work more
+consistent.
+
+
+Dask-SSH
+--------
+
+According to our user survey, the most common deployment mechanism was still
+SSH.  Dask has had a [command line dask-ssh
+tool](https://docs.dask.org/en/latest/setup/ssh.html#command-line) to make it
+easier to deploy with SSH for some time.  We recently updated this to also
+include a Python interface, which provides more control.
+
+```python
+>>> from dask.distributed import Client, SSHCluster
+>>> cluster = SSHCluster(
+...     ["host1", "host2", "host3", "host4"],
+...     connect_options={"known_hosts": None},
+...     worker_options={"nthreads": 2},
+...     scheduler_options={"port": 0, "dashboard_address": ":8797"}
+... )
+>>> client = Client(cluster)
+```
+
+This isn't what we recommend for large institutions, but it can be helpful for
+more informal groups who are just getting started.
+
+
+Dask-Jobqueue and Dask-Kubernetes Rewrite
+-----------------------------------------
+
+We've rewritten Dask-Jobqueue for SLURM/PBS/LSF/SGE cluster managers typically
+found in HPC centers and Dask-Kubernetes.  These now share a common codebase
+along with Dask SSH, and so are much more consistent and hopefully bug free.
+
+Hopefully users shouldn't notice much difference with existing workloads,
+but new features like asynchronous operation, integration with the Dask
+JupyterLab extension, and so on are more consistently available.  Also, we've
+been able to unify development and reduce our maintenance burden considerably.
+
+
+Dask-CloudProvider
+------------------
+
+For cloud deployments we generally recommend using a hosted Kubernetes or Yarn
+service, and then using Dask-Kubernetes or Dask-Yarn on top of these.
+
+However in some institutions these hosted services aren't yet accessible, and
```

Comment:
I would maybe reword this as something like "in some institutions they have made decisions or commitments to use certain vendor specific technologies."

I think a lot of the use cases for dask-cloudprovider are going to be folks who have gone "all in" on a certain cloud provider or set of technologies. I've seen a few organisations to this, sometimes for a reduced price, sometimes to encourage a constrained pallete of tools, etc.